### DISCUSSION TITLE
Cefsharp LoadError is triggered before the download is complete

### DISCUSSION THREAD

**Discussion by NaveenK-Philips** [cefsharp/CefSharp#3692] [General]
HI,

Facing one issue. 

Cefsharp LoadError is triggered before the download is complete. what is the issue?

Here is the code.

**DownloadHandler:**
                   var folderBrowser = new FolderBrowserDialog();
                    folderBrowser.WindowStartupLocation = System.Windows.WindowStartupLocation.CenterScreen;

                    if (folderBrowser.ShowDialog() == true)
                    {
                        if (!callback.IsDisposed)
                        {
                            using (callback)
                            {
                                var downloadPath = Path.Combine(folderBrowser.SelectedPath, downloadItem.SuggestedFileName);
                                callback.Continue(downloadPath, showDialog: false);
                            }
                        }
                    }

**LifeSpanHandler:**
  public bool OnBeforePopup(IWebBrowser chromiumWebBrowser, IBrowser browser, IFrame frame, string targetUrl, string targetFrameName, WindowOpenDisposition targetDisposition, bool userGesture, IPopupFeatures popupFeatures, IWindowInfo windowInfo, IBrowserSettings browserSettings, ref bool noJavascriptAccess, out IWebBrowser newBrowser)
        {
            browser.MainFrame.LoadUrl(targetUrl);
            newBrowser = null;
            return true;
        }

**Comment by amaitland** [comment]
> Cefsharp LoadError is triggered before the download is complete. what is the issue?

And what is the error code? If the error is Aborted then this is expected. Just ignore the Aborted error.

```c#
if (args.ErrorCode == CefErrorCode.Aborted)
{
  return;
}
```

**Reply by amaitland** [reply]
In future please format your code see https://www.markdownguide.org/extended-syntax/#fenced-code-blocks

**Reply by NaveenK-Philips** [reply]
**Error Text:** ERR_ABORTED,  **Error code:**  Aborted

Is this error code "Aborted" related to only for download since i have loaderror method to show simple html page on LoadError.

Code:
private void BrowserLoadError(object sender, LoadErrorEventArgs eventArgs)
        {
            var html = GetUrlNotAccessibleMessage();
            WebBrowser.LoadHtml(html);
        }

**Reply by amaitland** [reply]
Please format your code see  https://www.markdownguide.org/extended-syntax/#fenced-code-blocks

Simplify add the code I provided and ignore the aborted error. I have yet to see a case where it was relevant to show the user an error page for an aborted error.

**Reply by NaveenK-Philips** [reply]
Sure Alex Maitland. Thanks for your quick response. saved my time:)

**Reply by NaveenK-Philips** [reply]
Hi Alex,

What I observed is, Im able to download the image, still the LoadError is triggered(FolderBrowserDialog is open to select location to download the image). I did not understand what is the roor cuase for this?

Im currently using CefSharp WPF version 84.4.10.0 and avialable version in nuget is 91.1.211

Is this issue fixed in latest vesrion? if not then when this issue will be fixed.

Thanks,
Naveen

**Reply by amaitland** [reply]
> Im able to download the image, still the LoadError is triggered(FolderBrowserDialog is open to select location to download the image). I did not understand what is the roor cuase for this?

As I said having LoadError fire when a download starts is expected. This is not a bug, it's how chromium behaves internally. 

Have you updated your code to return when LoadError is called with Aborted?

**Reply by NaveenK-Philips** [reply]
Yes, I have updated the code.  Thanks Alex
